ingenialink/ethercat/network.py

```diff
@@ -455,56 +450,93 @@ def load_firmware(
         if password is None:
             password = self.DEFAULT_FOE_PASSWORD
 
-        arch = platform.architecture()[0]
-        sys_name = sys.platform
-        app_path = self.FOE_APPLICATION.get(sys_name, {}).get(arch, None)
-        if app_path is None:
-            raise NotImplementedError(
-                "Load FW by ECAT is not implemented for this OS and architecture:"
-                f" {sys_name} {arch}"
-            )
-        exec_path = os.path.join(os.path.dirname(inspect.getfile(bin_module)), app_path)
-        logger.debug(f"Call FoE application for {sys_name}-{arch}")
-        if sys_name == "linux":
-            try:
-                subprocess.run(
-                    f"chmod 777 {exec_path}",
-                    check=True,
-                    shell=True,
-                    encoding="utf-8",
-                )
-            except subprocess.CalledProcessError as e:
-                raise ILFirmwareLoadError("Could not change the FoE binary permissions.") from e
-        try:
-            if sys_name == "linux":
-                subprocess.run(
-                    f"{exec_path} {self.interface_name} {slave_id} {fw_file} "
-                    f"{int(boot_in_app)} {password}",
-                    check=True,
-                    shell=True,
-                    encoding="utf-8",
-                )
+        self._validate_slave_id(slave_id)
```

Comment:
We should not restart the master if the slave is not connected.
Starting the master is acceptable.
No worries if we "duplicate" a bit of code.

Place this here and leave the connect_to_slave as it was.
````
if not isinstance(slave_id, int) or slave_id < 0:
    raise ValueError("Invalid slave ID value")
if not self.__is_master_running:
    self._start_master()
    self.__init_nodes()
if len(self.__last_init_nodes) == 0:
    raise ILError("Could not find any slaves in the network.")
if slave_id not in self.__last_init_nodes:
    raise ILError(f"Slave {slave_id} was not found.")

````

@GullinBustin is going to work on a load_firmware api review soon and this may be simplified INGM-562

Reply:
Done